scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -314,3 +314,36 @@ def test_rotvec_calc_pipeline():
         [-3e-4, 3.5e-4, 7.5e-5]
         ])
     assert_allclose(Rotation.from_rotvec(rotvec).as_rotvec(), rotvec)
+
+
+def test_from_euler_single_rotation():
+    quat = Rotation.from_euler('z', 90, degrees=True).as_quaternion()
+    expected_quat = np.array([0, 0, 1, 1]) / np.sqrt(2)
+    assert_allclose(quat, expected_quat)
+
+
+def test_single_intrinsic_extrinsic_rotation():
+    ext = Rotation.from_euler('z', 90, degrees=True).as_dcm()
+    int = Rotation.from_euler('Z', 90, degrees=True).as_dcm()
+    assert_allclose(ext, int)
+
+
+def test_from_euler_rotation_order():
+    # Intrinsic rotation is same as extrinsic with order reversed
+    np.random.seed(0)
+    a = np.random.randint(low=0, high=180, size=(6, 3))
+    b = a[:, np.argsort([2, 1, 0])]
+    x = Rotation.from_euler('xyz', a, degrees=True).as_quaternion()
+    y = Rotation.from_euler('ZYX', b, degrees=True).as_quaternion()
+    assert_allclose(x, y)
+
+
+def test_from_euler_elementary_extrinsic_rotation():
```

Comment:
I suggest to test for some angles (like 30, 60, 45 degrees) common rotations 3-1-3 and 3-1-2 or 3-2-). You can find the DCM for them in many places.